desc/objectives/_geometry.py

```diff
@@ -707,22 +728,265 @@ def compute(self, equil_params, surface_params=None, constants=None):
         )
         plasma_coords = rpz2xyz(jnp.array([data["R"], data["phi"], data["Z"]]).T)
         if self._surface_fixed:
-            surface_coords = constants["surface_coords"]
+            data_surf = constants["data_surf"]
         else:
-            surface_coords = compute_fun(
+            data_surf = compute_fun(
                 self._surface,
                 self._surface_data_keys,
                 params=surface_params,
                 transforms=constants["surface_transforms"],
                 profiles={},
                 basis="xyz",
-            )["x"]
-        d = safenorm(plasma_coords[:, None, :] - surface_coords[None, :, :], axis=-1)
+            )
+
+        surface_coords = data_surf["x"]
+        diff_vec = plasma_coords[:, None, :] - surface_coords[None, :, :]
+        d = safenorm(diff_vec, axis=-1)
 
         if self._use_softmin:  # do softmin
             return jnp.apply_along_axis(softmin, 0, d, self._alpha)
         else:  # do hardmin
-            return d.min(axis=0)
+            if not self._use_signed_distance:
+                return d.min(axis=0)
+            else:
+                min_inds = d.argmin(axis=0, keepdims=True)
+                min_ds = jnp.take_along_axis(d, min_inds, axis=0).squeeze()
```

Comment:
this is the smallest magnitude distance, is that what we want? IE, suppose the plasma surface is bulging really far outside of the bounding surface, then this might still be small (as there is somewhere where the two surfaces touch) but what we want to penalize is the large negative distance where its bulging out the most.

Reply:
it is the smallest magnitude distance per point on the surface that we are considering, so even if the surfaces touch somewhere.. hm yea I see what you are saying. What I am doing would make sense/work if we were taking the minimum distances from the points on the plasma to the surface (i.e. if `dim_f = plas_grid.num_nodes`) as then we would be looking at the minimum distances per point on the plasma, and so if the plasma bulges far we would still have a point whose min distance is large and then it would be assigned a negative value based off the normal vector dot product.